ledger/src/shred.rs

```diff
@@ -394,15 +451,28 @@ impl Shred {
         }
     }
 
-    pub fn reference_tick_from_data(data: &[u8]) -> u8 {
-        let flags = data[SIZE_OF_COMMON_SHRED_HEADER + SIZE_OF_DATA_SHRED_HEADER - size_of::<u8>()];
+    pub fn reference_tick_from_data(data: &[u8], slot: Slot) -> u8 {
+        let size_of_data_shred_header = Shredder::get_expected_data_header_size_from_slot(slot);
+        let flags = data[SIZE_OF_COMMON_SHRED_HEADER + size_of_data_shred_header - size_of::<u8>()];
         flags & SHRED_TICK_REFERENCE_MASK
     }
 
     pub fn verify(&self, pubkey: &Pubkey) -> bool {
         self.signature()
             .verify(pubkey.as_ref(), &self.payload[SIZE_OF_SIGNATURE..])
     }
+
+    pub fn is_nonce_unlocked(slot: Slot) -> bool {
+        slot > UNLOCK_NONCE_SLOT
```

Comment:
@carllin Maybe this should be `>=`? Otherwise, strictly speaking the first slot in the epoch 47 isn't enabled

```rust
 pub const UNLOCK_NONCE_SLOT: Slot = 14_780_256; 
```

```
$ ../solana-cli  --url http://tds.solana.com:8899 epoch-info
Slot: 13686911
Epoch: 44
Epoch slot range: [13484256..13916256)
Epoch completed percent: 46.911%
Epoch completed slots: 202655/432000 (229345 remaining)
Epoch completed time: 22h 31m 2s/2days (1day 1h 28m 58s remaining)
$ pry
[5] pry(main)> 13484256 + 432000 * 3
=> 14780256
```

Reply:
@ryoqun, true but it's not a strict cutoff anyways, just needs to be in the ballpark :D

Reply:
@carllin OK! Sounds good. :)